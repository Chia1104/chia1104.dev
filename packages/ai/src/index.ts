import OpenAI from "openai";
import type { ClientOptions } from "openai";

export const createOpenAI = (options?: Partial<ClientOptions>) => {
  return new OpenAI({
    apiKey: process.env.OPENAI_API_KEY ?? "st-SECRETKEY",
    ...options,
  });
};
