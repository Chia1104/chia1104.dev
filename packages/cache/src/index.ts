export const name = "cache";
export * from "./rate-limiter";
export * from "./env";
export * from "./create-client";
export * from "./create-redis";
export * from "./create-upstash";
export * from "./with-rate-limiter";
