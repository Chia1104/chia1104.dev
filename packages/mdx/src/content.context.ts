"use client";

import { createContext, useContext } from "react";

import type { ContentProps } from "./types";

export const ContentContext = createContext<ContentProps>({} as ContentProps);

export const useContent = () => {
  const context = useContext(ContentContext);
  if (!context) {
    throw new Error("useContent must be used within a ContentProvider");
  }
  return context;
};
