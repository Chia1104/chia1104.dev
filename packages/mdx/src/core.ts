import {
  compileMDX as _compileMDX,
  type MDXRemoteProps,
} from "next-mdx-remote/rsc";
import { serialize } from "next-mdx-remote/serialize";
import remarkGfm from "remark-gfm";
import rehypeSlug from "rehype-slug";
import rehypeAutolinkHeadings from "rehype-autolink-headings";
import rehypePrettyCode from "rehype-pretty-code";
import rehypePrism from "rehype-prism-plus/generator";

export const serializeMDX = async (
  content: string,
  options?: MDXRemoteProps
) => {
  return await serialize(content, {
    mdxOptions: {
      parseFrontmatter: false,
      remarkPlugins: [
        [remarkGfm, { singleTilde: false }],
        ...(options?.options?.mdxOptions?.remarkPlugins ?? []),
      ],
      rehypePlugins: [
        [rehypeSlug],
        [rehypePrism, { ignoreMissing: true }],
        [rehypeAutolinkHeadings],
        [
          // @ts-ignore
          rehypePrettyCode,
          {
            theme: {
              light: "github-light",
              dark: "github-dark",
            },
            keepBackground: false,
          },
        ],
        ...(options?.options?.mdxOptions?.rehypePlugins ?? []),
      ],
      ...options?.options?.mdxOptions,
    },
    ...options,
  });
};

export const compileMDX = async (content: string, options?: MDXRemoteProps) => {
  return await _compileMDX({
    ...options,
    source: content,
    options: {
      mdxOptions: {
        parseFrontmatter: false,
        remarkPlugins: [
          [remarkGfm, { singleTilde: false }],
          ...(options?.options?.mdxOptions?.remarkPlugins ?? []),
        ],
        rehypePlugins: [
          [rehypeSlug],
          [rehypePrism, { ignoreMissing: true }],
          [rehypeAutolinkHeadings],
          [
            // @ts-ignore
            rehypePrettyCode,
            {
              theme: {
                light: "github-light",
                dark: "github-dark",
              },
              keepBackground: false,
            },
          ],
          ...(options?.options?.mdxOptions?.rehypePlugins ?? []),
        ],
        ...options?.options?.mdxOptions,
      },
      ...options?.options,
    },
  });
};
