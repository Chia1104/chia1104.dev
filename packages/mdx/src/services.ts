import { cache } from "react";

import { compileMDX as _compileMDX } from "@fumadocs/mdx-remote";

import type { schema } from "@chia/db";
import { ContentType } from "@chia/db/types";

import { FumadocsComponents, V1MDXComponents } from "./mdx-components";
import type { ContentProps } from "./types";

export const compileMDX = cache(async (content: string) => {
  return _compileMDX({
    source: content,
    components: {
      ...FumadocsComponents,
      ...V1MDXComponents,
    },
  });
});

export const getContentProps = async ({
  contentType,
  content,
}: {
  contentType: ContentType;
  content: Partial<
    Pick<schema.Content, "content" | "source" | "unstable_serializedSource">
  >;
}) => {
  switch (contentType) {
    case ContentType.Mdx: {
      const compiled = await compileMDX(content.content ?? "");
      return {
        type: ContentType.Mdx,
        toc: compiled.toc,
        content: compiled.content,
      } satisfies ContentProps;
    }
    default: {
      return {
        type: contentType,
        content: content.content,
      } satisfies ContentProps;
    }
  }
};
