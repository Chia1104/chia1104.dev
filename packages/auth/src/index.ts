import { betterAuth } from "better-auth";
import { drizzleAdapter } from "better-auth/adapters/drizzle";
import { magicLink } from "better-auth/plugins";
import { Resend } from "resend";

import { createRedis } from "@chia/cache";
import { connectDatabase } from "@chia/db/client";
import * as schemas from "@chia/db/schema";
import { Role } from "@chia/db/types";
import EmailTemplate from "@chia/ui/features/AuthEmailTemplate";
import { CONTACT_EMAIL } from "@chia/utils";

import { env } from "./env";
import { useSecureCookies, getCookieDomain } from "./utils";

export const name = "auth-core";

const resend = new Resend(env.RESEND_API_KEY);
const database = await connectDatabase();
const redis = createRedis();

const getOrigin = (url?: string) => {
  if (!url) {
    return undefined;
  }
  return new URL(url).origin;
};

export const auth = betterAuth({
  appName: "Chia1104.dev",

  socialProviders: {
    github:
      env.GITHUB_CLIENT_ID && env.GITHUB_CLIENT_SECRET
        ? {
            clientId: env.GITHUB_CLIENT_ID,
            clientSecret: env.GITHUB_CLIENT_SECRET,
          }
        : undefined,
    google:
      env.GOOGLE_CLIENT_ID && env.GOOGLE_CLIENT_SECRET
        ? {
            clientId: env.GOOGLE_CLIENT_ID,
            clientSecret: env.GOOGLE_CLIENT_SECRET,
          }
        : undefined,
  },

  /**
   * database adapter
   */
  database: drizzleAdapter(database, {
    provider: "pg",
    schema: {
      ...schemas,
      user: schemas.users,
    },
  }),

  user: {
    additionalFields: {
      role: {
        type: [Role.User, Role.Admin],
        required: true,
        defaultValue: Role.User,
        input: true,
      },
    },
  },

  secondaryStorage: {
    get: async (key) => {
      const value = await redis.get(key);
      return value ? value : null;
    },
    set: async (key, value, ttl) => {
      if (ttl) await redis.set(key, value, "EX", ttl);
      else await redis.set(key, value);
    },
    delete: async (key) => {
      await redis.del(key);
    },
  },

  /**
   * base path for all auth routes
   */
  basePath: "/api/v1/auth",

  baseURL: getOrigin(env.AUTH_URL),
  secret: env.AUTH_SECRET,

  /**
   * advanced configuration
   */
  advanced: {
    cookiePrefix: "chia1104.dev",
    crossSubDomainCookies: {
      enabled: true,
      domain: getCookieDomain({ env }),
    },
    defaultCookieAttributes: {
      secure: useSecureCookies,
    },
  },

  trustedOrigins: env.CORS_ALLOWED_ORIGIN
    ? env.CORS_ALLOWED_ORIGIN.split(",")
    : [],

  plugins: [
    magicLink({
      sendMagicLink: async ({ email, url }) => {
        await resend.emails.send({
          from: CONTACT_EMAIL,
          to: email,
          subject: "Sign in to Chia1104.dev",
          text: "Please click the link below to sign in",
          react: EmailTemplate({
            url,
            host: new URL(url).host,
          }),
        });
      },
    }),
  ],
});
