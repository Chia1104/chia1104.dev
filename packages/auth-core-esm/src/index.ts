import { Auth as InternalAuth } from "@auth/core";
import type { AuthConfig } from "@auth/core";
import Github from "@auth/core/providers/github";
import Google from "@auth/core/providers/google";
import { DrizzleAdapter } from "@auth/drizzle-adapter";

import { db, localDb, betaDb, schema } from "@chia/db";
import { getDb } from "@chia/utils";

import { env } from "./env";
import { getBaseConfig } from "./utils";

declare module "@auth/core/types" {
  interface Session extends DefaultSession {
    user: {
      role: "admin" | "user";
    } & DefaultSession["user"];
  }

  interface User {
    role: "admin" | "user";
  }
}

const AUTH_URL = env.AUTH_URL?.replace(/\/api\/auth$/, "");

export const name = "auth-core";

export const getConfig = (
  req?: Request,
  config?: Partial<Omit<AuthConfig, "raw">>
) => {
  return {
    ...getBaseConfig({
      req,
      env: {
        AUTH_URL,
        AUTH_COOKIE_DOMAIN: env.AUTH_COOKIE_DOMAIN,
        AUTH_SECRET: env.AUTH_SECRET,
      },
    }),
    adapter: DrizzleAdapter(
      getDb(undefined, {
        db,
        betaDb,
        localDb,
      }),
      {
        usersTable: schema.users,
        accountsTable: schema.accounts,
        sessionsTable: schema.sessions,
        verificationTokensTable: schema.verificationTokens,
      }
    ),
    providers: [
      Google({
        clientId: env.GOOGLE_CLIENT_ID,
        clientSecret: env.GOOGLE_CLIENT_SECRET,
      }),
      Github({
        clientId: env.GITHUB_CLIENT_ID,
        clientSecret: env.GITHUB_CLIENT_SECRET,
      }),
    ],
    ...config,
  } satisfies AuthConfig;
};

export const Auth = (
  request: Request,
  config?: Partial<Omit<AuthConfig, "raw">>
) => InternalAuth(request, getConfig(request, config));
export type { Session } from "@auth/core/types";
