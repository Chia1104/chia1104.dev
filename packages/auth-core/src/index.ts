import { Auth as InternalAuth } from "@auth/core";
import type { AuthConfig } from "@auth/core";
import Github from "@auth/core/providers/github";
import Google from "@auth/core/providers/google";

import { createRedis } from "@chia/cache";
import { connectDatabase } from "@chia/db/client";

import { adapter } from "./adapter";
import { env } from "./env";
import { getBaseConfig, SESSION_MAX_AGE, SESSION_UPDATE_AGE } from "./utils";

declare module "@auth/core/types" {
  interface Session extends DefaultSession {
    user: {
      role: "admin" | "user";
    } & DefaultSession["user"];
  }

  interface User {
    role: "admin" | "user";
  }
}

const AUTH_URL = env.AUTH_URL?.replace(/\/api\/auth$/, "");

export const name = "auth-core";

export const getConfig = async (
  req?: Request,
  config?: Partial<Omit<AuthConfig, "raw">>
) => {
  return {
    ...getBaseConfig({
      req,
      env: {
        AUTH_URL,
        AUTH_COOKIE_DOMAIN: env.AUTH_COOKIE_DOMAIN,
        AUTH_SECRET: env.AUTH_SECRET,
      },
    }),
    adapter: adapter({
      db: await connectDatabase(),
      redis: createRedis(),
    }),
    providers: [
      Google({
        clientId: env.GOOGLE_CLIENT_ID,
        clientSecret: env.GOOGLE_CLIENT_SECRET,
      }),
      Github({
        clientId: env.GITHUB_CLIENT_ID,
        clientSecret: env.GITHUB_CLIENT_SECRET,
      }),
    ],
    session: {
      maxAge: SESSION_MAX_AGE,
      updateAge: SESSION_UPDATE_AGE,
    },
    ...config,
  } satisfies AuthConfig;
};

export const Auth = (
  request: Request,
  config?: Partial<Omit<AuthConfig, "raw">>
) => (async () => InternalAuth(request, await getConfig(request, config)))();
export type { Session } from "@auth/core/types";
