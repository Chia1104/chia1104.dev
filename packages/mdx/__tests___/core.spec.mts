import { compileMDX } from "../src/core";

describe("compileMDX", () => {
  it("should compile mdx", async () => {
    const content = `
    # Hello
    `;
    // const result = await compileMDX(content);
    // console.log(result.content);
  });
});
