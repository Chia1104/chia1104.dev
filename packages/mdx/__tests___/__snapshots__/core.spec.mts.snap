// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compileMDX > should compile mdx 1`] = `
{
  "content": <MDXContent
    components={{}}
  />,
  "frontmatter": {},
}
`;
